docs/migrating_versions/migrating_to_v0_5_0.rst

```diff
@@ -0,0 +1,134 @@
+The key advantages of the new API are:
+
+    - Support for evalulating multiple metric functions at once
+    - Support for multiple sensitive features
+    - Support for control features
+
+The :class:`MetricFrame` class has a constructor similar to ``group_summary()``.
+In v0.4.6, one would write
+
+.. code-block::
+
+    gs = group_summary(metric_func, y_true, y_pred, sensitive_features=A_col)
+
+With the new API, this becomes
+
+.. code-block::
+
+    mf = MetricFrame(metric_func, y_true, y_pred, sensitive_features=A_col)
+
+The new object has :attr:`.MetricFrame.overall` and :attr:`.MetricFrame.by_group`
+properties, to access the metric evaluated on the entire dataset, and the metric
+evaluated on the subgroups of ``A_col``.
+
+In v0.4.6, we provided the following aggregator functions to compute a single scalar
+from the result of ``group_summary()``.
+
+    - ``group_min_from_summary()``
+    - ``group_max_from_summary()``
+    - ``difference_from_summary()``
+    - ``ratio_from_summary()``
+
+With :class:`.MetricFrame` these become methods:
+
+    - :meth:`.MetricFrame.group_min`
+    - :meth:`.MetricFrame.group_max`
+    - :meth:`.MetricFrame.difference`
+    - :meth:`.MetricFrame.ratio`
+
+Before, one might write:
+
+.. code-block::
+
+    min_by_group = group_min_from_summary(gs)
+
+Now, one can write:
+
+.. code-block::
+
+    min_by_group = mf.group_min()
+
+There is a ``method=`` argument to :meth:`.MetricFrame.difference`
+and :meth:`.MetricFrame.ratio` which can be set to ``between_groups``
+and ``to_overall``. To obtain behaviour similar to:
+
+.. code-block::
+
+    diff = difference_from_summary(gs)
+
+use
+
+.. code-block::
+
+    diff = mf.difference(method='between_groups')
+
+The ``to_overall`` alternative will evaluate the differences (or ratios)
+relative to the overall value, rather than just between the groups identified
+by the sensitive feature.
+
+The ``make_derived_metric()`` function has been removed, but will be reintroduced
+in a future release. The pregenerated functions such as ``accuracy_score_group_min()``
+and ``precision_score_difference()`` remain.
+
+For an introduction to all the new features, see the 
+:ref:`sphx_glr_auto_examples_plot_new_metrics.py` example in
+:ref:`sphx_glr_auto_examples`.
+
+
+Renaming of members
+-------------------
+
+We have renamed a number of class members from ``_<name>`` to ``<name>_``.
+For example in both :class:`.ExponentiatedGradient` and :class:`.GridSearch`,
+the ``_predictors`` member is now called ``predictors_``.
+
+
+Exponentiated Gradient and Moments
+----------------------------------
+
+In addition to the trailing underscore change mentioned above, several
+adjustments have been made to :class:`.ExponentiatedGradient`.
+The ``T`` argument has been renamed to ``max_iter``, and the ``eta_mul``
+argument to ``eta0``.
+
+Furthermore, the ``eps`` argument was previously being used for two
```

Comment:
Let's update the description to something like this:

```
Furthermore, the ``eps`` argument was previously used for two
different purposes, and these two uses have now been separated.
The use of ``eps`` as the righthand side of the constraints
has now been moved to the :class:`.Moment` classes.
The only remaining use of the ``eps`` argument
is to control the optimality requirements for the optimization
algorithm in :class:`.ExponentiatedGradient`.
```

Reply:
Done.... although I have certain reservations about "the optimality of the optimization"